docs/docs.nix

```diff
@@ -79,6 +79,37 @@
             '';
           };
         };
+        generate-rust-docs = {
+          type = "app";
+          program = pkgs.writeShellApplication {
+            name = "generate-rust-docs";
+            text =
+              let
+                /**
+                 * The list of rust packages that we want to generate docs for.
+                 */
+                rustPackages = [ "hubble" "voyager" ];
+              in
+              ''
+                ${ensureAtRepositoryRoot}
+                
+                rm -rf docs/generated/rust
+                mkdir -p docs/generated/rust
+
+                ${lib.concatMapStrings (pkg: ''
+                  cargo rustdoc \
+                    --release \
+                    --all-features \
+                    --package='${pkg}' \
```

Comment:
you can pass -p more than once at a time. use https://doc.rust-lang.org/nightly/rustdoc/unstable-features.html#--enable-index-page-generate-a-default-index-page-for-docs as well for a root level directory with all packages.